web/packages/teleport/src/HeadlessSso/HeadlessSso.tsx

```diff
@@ -0,0 +1,127 @@
+/**
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+import React, { useEffect, useState } from 'react';
+
+import { Spinner } from 'design/Icon';
+
+import auth from 'teleport/services/auth';
+import { useParams } from 'teleport/components/Router';
+import HeadlessSsoDialog from 'teleport/components/HeadlessSsoDialog/HeadlessSsoDialog';
+import { CardAccept, CardDenied } from 'teleport/HeadlessSso/Cards';
+
+export function HeadlessSso() {
+  const { requestId } = useParams<{ requestId: string }>();
+
+  const [state, setState] = useState({
+    ipAddress: '',
+    status: 'pending',
+    errorText: '',
+    publicKey: null as PublicKeyCredentialRequestOptions,
+  });
+
+  useEffect(() => {
+    const setIpAddress = (response: { clientIpAddress: string }) => {
+      setState({
+        ...state,
+        status: 'loaded',
+        ipAddress: response.clientIpAddress,
+      });
+    };
+
+    auth
+      .headlessSSOGet(requestId)
+      .then(setIpAddress)
+      .catch(e => {
+        setState({
+          ...state,
+          status: 'error',
+          errorText: e.toString(),
+        });
+      });
+  }, [requestId]);
+
+  const setSuccess = () => {
+    setState({ ...state, status: 'success' });
+  };
+
+  const setRejected = () => {
+    setState({ ...state, status: 'rejected' });
+  };
+
+  if (state.status == 'pending') {
+    // Fix the spinner location.
+    return (
+      <Spinner
+        css={`
+          vertical-align: top;
+        `}
+      />
+    );
```

Comment:
@hatched Can we get your help here? The spinner is at the top of the page currently. I want to have it in the center.